tests/test_example_repo.py

```diff
@@ -123,3 +123,26 @@ def test_no_git_no_warning(capsys):
         os.environ['PATH'] = path_backup
     for k, v in data.items():
         assert v == ['None', 'undefined']
+
+
+def test_exclude_commits_dates():
+    data = run_sphinx(
+        'repo_full',
+        git_exclude_commits='6bb90c6027c3788d3891f833f017dbf8d229e432')
+    assert data == {
+        **expected_results,
+        'api': [time1, 'defined'],
+        'example_module.example_function': [time1, 'undefined'],
+    }
+
+
+def text_exclude_commits_warning(capsys):
```

Comment:
This test is broken, but it never runs because it has the wrong name:

```suggestion
def test_exclude_commits_warning(capsys):
```

When changing that, it turns out that the exception isn't raised as expected.

Reply:
Whoops, sorry for this

Reply:
Ok, so I now understand why the exception is not raised. This is because The loop is broken at this block:
https://github.com/mgeier/sphinx-last-updated-by-git/blob/69dbf2c91cedec1465e9aad868b49e0e6d30fe31/src/sphinx_last_updated_by_git.py#L123-L126

Since "added but uncommited" files are _in fine_ completely ignored, I would suggest to ignore them from the beginning by applying this patch:
```diff
diff --git a/src/sphinx_last_updated_by_git.py b/src/sphinx_last_updated_by_git.py
index aa42aaf..a9b45b5 100644
--- a/src/sphinx_last_updated_by_git.py
+++ b/src/sphinx_last_updated_by_git.py
@@ -34,7 +34,7 @@ def update_file_dates(git_dir, exclude_commits, file_dates):
 
     existing_files = subprocess.check_output(
         [
-            'git', 'ls-files', '-z', '--', *requested_files
+            'git', 'ls-tree', '--name-only', '-z', 'HEAD', '--', *requested_files
         ],
         cwd=git_dir,
         stderr=subprocess.PIPE,
@@ -66,10 +66,6 @@ def parse_log(stream, requested_files, git_dir, exclude_commits, file_dates):
     requested_files = set(f.encode('utf-8') for f in requested_files)
 
     line0 = stream.readline()
-    if not line0:
-        logger.info('added but uncommitted file(s) in {}: {}'.format(
-            git_dir, {f.decode('utf-8') for f in requested_files}))
-        return
 
     # First line is blank
     assert not line0.rstrip(), 'unexpected git output in {}: {}'.format(
@@ -120,11 +116,6 @@ def parse_log(stream, requested_files, git_dir, exclude_commits, file_dates):
             else:
                 file_dates[file.decode('utf-8')] = timestamp, too_shallow
 
-        if requested_files and not parent_commits:
-            logger.info('added but uncommitted file(s) in {}: {}'.format(
-                git_dir, {f.decode('utf-8') for f in requested_files}))
-            break
-
 
 def _env_updated(app, env):
     # NB: We call git once per sub-directory, because each one could
```

With `git ls-tree HEAD` only commited files are listed. So there is no need for both the checks removed. I tested I by manually adding files to the staged area but it would be good to have a test for this.

With this patch, The only way that the parsing loop can go out of lines is here:
https://github.com/mgeier/sphinx-last-updated-by-git/blob/69dbf2c91cedec1465e9aad868b49e0e6d30fe31/src/sphinx_last_updated_by_git.py#L80-L88

Reply:
I extracted this comment into a new Pull Request: #46